batch/batch/cloud/gcp/worker/disk.py

```diff
@@ -109,9 +109,20 @@ async def _attach(self):
                 'deviceName': self.name,
             }
 
-            self.last_response = await self.compute_client.attach_disk(
-                f'/zones/{self.zone}/instances/{self.instance_name}/attachDisk', json=config
-            )
+            try:
+                self.last_response = await self.compute_client.attach_disk(
+                    f'/zones/{self.zone}/instances/{self.instance_name}/attachDisk', json=config
+                )
+            except aiogoogle.client.compute_client.GCPOperationError as e:
+                if e.status == 400:
+                    assert e.error_messages and e.error_codes
+                    if all(self.instance_name in em for em in e.error_messages) and all(
+                        em == 'RESOURCE_IN_USE_BY_ANOTHER_RESOURCE' for em in e.error_codes
+                    ):
```

Comment:
nit: can we extract this check into some helper like `_disk_is_already_attached_to_this_machine` and then turn this into `if not …: raise` instead of the `pass`? I think this is correct but I find it hard to parse as is.